### ISSUE TITLE
incorrect component.Type tutorial guidance since OTEL v0.97.0

### ISSUE THREAD

**Issue by mjnowen** [open-telemetry/opentelemetry.io#4263]
Due to OTEL v0.97.0 release and specifically this PR: https://github.com/open-telemetry/opentelemetry-collector/pull/9472/files

These docs/tutorial sections need updating to take into account `component.Type` is no longer a string and is now an opaque struct.

https://opentelemetry.io/docs/collector/building/receiver/#implementing-the-receiverfactory-interface
https://opentelemetry.io/docs/collector/building/receiver/#identifying-and-providing-default-settings
https://opentelemetry.io/docs/collector/building/receiver/#specifying-the-receivers-capabilities

Specifically, these lines in the above docs page need fixing:

```
`component.Type`: a unique string identifier for your receiver across all Collector’s components.
```

```
const (
	typeStr         = "tailtracer"
	defaultInterval = 1 * time.Minute
)
```

```
// NewFactory creates a factory for tailtracer receiver.
func NewFactory() receiver.Factory {
	return receiver.NewFactory(
		typeStr,
		createDefaultConfig,
		receiver.WithTraces(createTracesReceiver, component.StabilityLevelAlpha))
}
```

and finally:

```
package tailtracer

import (
	"context"
	"time"

	"go.opentelemetry.io/collector/component"
	"go.opentelemetry.io/collector/consumer"
	"go.opentelemetry.io/collector/receiver"
)

const (
	typeStr         = "tailtracer"
	defaultInterval = 1 * time.Minute
)

func createDefaultConfig() component.Config {
	return &Config{
		Interval: string(defaultInterval),
	}
}

func createTracesReceiver(_ context.Context, params receiver.CreateSettings, baseCfg component.Config, consumer consumer.Traces) (receiver.Traces, error) {
	return nil, nil
}

// NewFactory creates a factory for tailtracer receiver.
func NewFactory() receiver.Factory {
	return receiver.NewFactory(
		typeStr,
		createDefaultConfig,
		receiver.WithTraces(createTracesReceiver, component.StabilityLevelAlpha))
}
```

Suggested change:

```
const (
	defaultInterval = 1 * time.Minute
)

var (
	compType       = component.MustNewType("tailtracer")
)
```

followed by:

```
// NewFactory creates a factory for tailtracer receiver.
func NewFactory() receiver.Factory {
	return receiver.NewFactory(
		compType,
		createDefaultConfig,
		receiver.WithTraces(createTracesReceiver, component.StabilityLevelAlpha))
}
```

**Comment by svrnm** [comment]
Thank you @mjnowen!

@open-telemetry/collector-approvers ptal

**Comment by mx-psi** [comment]
@mjnowen is correct, and their suggested change is what I would do :) 

@mjnowen Would you be up to filing a PR for updating this?

**Comment by mx-psi** [comment]
Looks like #4250 addresses this

**Comment by mjnowen** [comment]
Fixed with #4250

**Comment by theletterf** [comment]
Closing as fixed!
